src/tools/interop/idt/capture/platform/android/android.py

```diff
@@ -0,0 +1,220 @@
+                f"Detected connection string; attempting to connect: {connect_command}")
+            BashRunner(connect_command, sync=True, capture_output=False)
+            self.get_adb_devices()
+
+    def _device_id_user_input(self) -> None:
+        print('If there is no output below, press enter after connecting your phone under test OR')
+        print('Enter (copy paste) the target device id from the list of available devices below OR')
+        print('Enter $IP4:$PORT to connect wireless debugging.')
+        self._log_adb_devices()
+        temp_device_id = input('').strip()
+        self._check_connect_wireless_adb(temp_device_id)
+        if self._only_one_device_connected():
+            self._set_device_if_only_one_connected()
+        elif temp_device_id not in self.adb_devices:
+            print('Entered device not in adb devices!')
+        else:
+            self.device_id = temp_device_id
+
+    def _choose_device_id(self) -> None:
+        """
+        Prompts the user to select a single device ID for this transport
+        If only one device is ever connected, use it.
+        """
+        self._set_device_if_only_one_connected()
+        while self.device_id not in self.get_adb_devices():
+            self._device_id_user_input()
+        print(f'Selected device {self.device_id}')
+
+    def _authorize_adb(self) -> None:
+        """
+        Prompts the user until a single device is selected and adb is auth'd
+        """
+        self.get_adb_devices()
+        self._choose_device_id()
+        while not self.get_adb_devices()[self.device_id]:
+            print('Confirming authorization, press enter after auth')
+            input('')
+        print(f'Target android device ID is authorized: {self.device_id}')
+
+    def check_screen(self) -> bool:
+        screen_cmd_output = self.run_adb_command(
+            self.check_screen_command, capture_output=True)
+        return "true" in screen_cmd_output.get_captured_output()
+
+    def ensure_screen_is_on(self) -> None:
+        screen_on = self.check_screen()
+        while not screen_on:
+            input(
+                "Please turn the screen on so screen recording can start, press enter once on!")
+            screen_on = self.check_screen()
+            if not screen_on:
+                print("Screen is still not on for recording!")
+
+    def start_streaming(self) -> None:
+        if self.screen_proc is None or not self.screen_proc.command_is_running():
+            self.ensure_screen_is_on()
+            self.screen_proc = BashRunner(self.screen_command)
+            print(f'Screen recording started {self.screen_cap_output_path}')
+            self.screen_recording_already_pulled = False
+        if self.logcat_proc is None or not self.logcat_proc.command_is_running():
+            self.logcat_proc = BashRunner(self.logcat_command)
+            print(f'logcat started {self.logcat_output_path}')
+
+    def pull_screen_recording(self) -> None:
```

Comment:
I think we should model the Android screen recording as it's own Monitor. Android devices in different states are going to support the monitoring commands to a different extent. For some phones or etc they are probably going to need to disable certain monitors. On my rooted phone, I would have access to all of the monitors. A QA with an off-the-shell phone that was never configured for debugging won't have certain things enabled, like HCI logging.

Reply:
Thanks I think it will be a good idea to break this up!